gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/KubernetesExtension.java

```diff
@@ -288,4 +301,11 @@ public void mapping(Closure<?> closure) {
     mappings.add(closureTo(closure, MappingConfig.class));
   }
 
+  public File getManifest(KitLogger kitLogger, KubernetesClient kubernetesClient, JavaProject javaProject) {
+    if (OpenshiftHelper.isOpenShift(kubernetesClient)) {
+      kitLogger.warn("OpenShift cluster detected, using Kubernetes manifests");
+      kitLogger.warn("Switch to openshift-gradle-plugin in case there are any problems");
+    }
+    return getKubernetesManifest().getOrElse(new File(javaProject.getBaseDirectory(), DEFAULT_KUBERNETES_MANIFEST));
```

Comment:
Create a getKubernetesManifestOrDefault method that can be reused in the OpenShiftExtension. This way we keep the default logic in a single place.